alchemist-maps/src/main/java/it/unibo/alchemist/boundary/gpsload/impl/TraceLoader.java

```diff
@@ -176,14 +180,17 @@ public TraceLoader(final String path,
         return Optional.ofNullable(cyclic ? null : traces.size());
     }
 
-    private static GPSTimeAlignment makeNormalizer(final String clazzName, final Object... args) {
+    private static GPSTimeAlignment makeNormalizer(final String clazzName, final Object... args) throws InterruptedException {
         final String fullName = clazzName.contains(".")
                 ? clazzName
                 : GPSTimeAlignment.class.getPackage().getName() + "." + clazzName;
         try {
             final Class<?> targetClass = ResourceLoader.classForName(fullName);
             if (GPSTimeAlignment.class.isAssignableFrom(targetClass)) {
-                return (GPSTimeAlignment) FACTORY.build(targetClass, args).getCreatedObjectOrThrowException();
+                SEMAPHORE.acquire();
```

Comment:
Use `acquireUninterruptibly()` and remove the exception propagation

Reply:
done